### PR TITLE
Add getAllLeaves method for the trie

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
Added getAllLeaves method to the trie that iterates the trie and returns a map[string][]byte containing all the information (key - value pairs) about the leafNodes.

### CODE REVIEW

File: data/trie/patriciaMerkleTrie.go [hunk 1]
```diff
@@ -303,3 +304,18 @@ func emptyTrie(root []byte) bool {
...
 	return false
 }
+
+// GetAllLeaves iterates the trie and returns a map that contains all leafNodes information
+func (tr *patriciaMerkleTrie) GetAllLeaves() (map[string][]byte, error) {
+	if tr.root == nil {
+		return map[string][]byte{}, nil
+	}
+
+	leafs := make(map[string][]byte)
```
Comment (bogdan-rosianu) [inline data/trie/patriciaMerkleTrie.go]: leaves ?

File: data/trie/leafNode.go [hunk 2]
```diff
@@ -272,3 +272,19 @@ func (ln *leafNode) deepClone() node {
 
 	return clonedNode
 }
+
+func (ln *leafNode) getAllLeaves(leafs map[string][]byte, key []byte, _ data.DBWriteCacher, _ marshal.Marshalizer) error {
```
Comment (bogdan-rosianu) [inline data/trie/leafNode.go]: leaves ?

File: data/trie/extensionNode.go [hunk 3]
```diff
@@ -417,3 +417,23 @@ func (en *extensionNode) deepClone() node {
 
 	return clonedNode
 }
+
+func (en *extensionNode) getAllLeaves(leafs map[string][]byte, key []byte, db data.DBWriteCacher, marshalizer marshal.Marshalizer) error {
```
Comment (bogdan-rosianu) [inline data/trie/extensionNode.go]: leaves ?

File: data/trie/branchNode.go [hunk 4]
```diff
@@ -537,3 +537,29 @@ func (bn *branchNode) deepClone() node {
 
 	return clonedNode
 }
+
+func (bn *branchNode) getAllLeaves(leafs map[string][]byte, key []byte, db data.DBWriteCacher, marshalizer marshal.Marshalizer) error {
```
Comment (bogdan-rosianu) [inline data/trie/branchNode.go]: leaves ?

File: data/trie/patriciaMerkleTrie_test.go [hunk 5]
```diff
@@ -387,6 +387,19 @@ func TestPatriciaMerkleTrie_DeepCloneShouldWork(t *testing.T) {
 	assert.Equal(t, originalRoot, clonedTrie)
 }
 
+func TestPatriciaMerkleTrie_GetAllLeafs(t *testing.T) {
+	t.Parallel()
+
+	tr := initTrie()
+	leafs, err := tr.GetAllLeaves()
```
Comment (bogdan-rosianu) [inline data/trie/patriciaMerkleTrie_test.go]: leaves ? also in the test name

File: data/trie/patriciaMerkleTrie.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie.go]: Changed.

File: data/trie/leafNode.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply data/trie/leafNode.go]: Changed.

File: data/trie/extensionNode.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply data/trie/extensionNode.go]: Changed.

File: data/trie/branchNode.go (hunk 4 above)
Comment (BeniaminDrasovean) [reply data/trie/branchNode.go]: Changed.

File: data/trie/patriciaMerkleTrie_test.go (hunk 5 above)
Comment (BeniaminDrasovean) [reply data/trie/patriciaMerkleTrie_test.go]: Changed.

**Pull request merged**